### PR TITLE
[development] Android compatibility fixes

### PR DESCRIPTION
**Pull request by AMZN-ScottR**
- Added support for NDK r23
- Added support for the versioned Android 'cmdline-tools' package which replace the deprecated 'tools' package
- Updated max Gradle version to latest point release of 7.0 to fix volume query bug during sync in some configurations
- Updated min Android Gradle plugin version to latest point release of 4.2
- Fixed issue with Gradle task chaining for some custom copy tasks
- Replaced use of deprecated 'jcenter' repo with 'mavenCentral' in Android project generator template
- Increased debug logging around adb calls in Android deployment script
- Fixed a couple try/except cases that were looking for the wrong exception when invoking `adb_*` functions
- Fixed issue where asset cache would get deleted when regenerating an existing Android Gradle project with `--overwrite-existing`
- Fixed issues with running Android project generation scripts on Unix systems

Signed-off-by: AMZN-ScottR <24445312+AMZN-ScottR@users.noreply.github.com>

### CODE REVIEW

File: cmake/Tools/common.py [hunk 1]
```diff
@@ -552,7 +558,7 @@ def exec(self, arguments, capture_stdout=False, suppress_stderr=False, cwd=None)
                 call_args.append(str(arguments))
             logging.debug("exec(%s)", subprocess.list2cmdline(call_args))
             result = subprocess.run(call_args,
-                                    shell=True,
+                                    shell=(platform.system() == 'Windows'),
```
Comment (spham-amzn) [inline cmake/Tools/common.py:561]: Do we need 'shell' anymore?

Comment (AMZN-ScottR) [reply cmake/Tools/common.py:561]: Windows might still require it but will confirm

Comment (AMZN-ScottR) [reply cmake/Tools/common.py:561]: Confirmed, still required for Windows.

**Review by amzn-sj — APPROVED** [review]

File: Code/Tools/Android/ProjectBuilder/root.build.gradle.in [hunk 2]
```diff
@@ -8,7 +8,7 @@
 buildscript {
     repositories {
         google()
-        jcenter()
+        mavenCentral()
```
Comment (spham-amzn) [inline Code/Tools/Android/ProjectBuilder/root.build.gradle.in:11]: yes! I been wanting to do this

**Review by spham-amzn — APPROVED** [review]

**Pull request merged**